--- Makefile	2012-10-22 11:11:09.858642053 -0400
+++ mphack.mk	2012-10-22 11:11:09.857642025 -0400
@@ -1,3 +1,9 @@
+# A HACKED LaTeX/MetaPost makefile -- requires GNU Make
+
+# Used when mpost is too old to support filenametemplate.
+
+# ---------
+
 # A LaTeX/MetaPost makefile for CPSA documents
 
 # Requrements
@@ -21,7 +27,9 @@
 MP_SRCS	= cpsadiagrams.mp carriers.mp termtree.mp
 
 %-0.mps:	%.mp
-	mpost -tex=latex $*
+	grep -v filenametemplate $*.mp > $*_tmp.mp
+	mpost -tex=latex $*_tmp
+	for i in $*_tmp.*[0-9]; do mv $$i `echo $$i | sed 's/_tmp\./-/g'`.mps; done
 
 %.bbl:	$(BIB).bib
 	if test ! -f $*.aux; then pdflatex $*.tex; fi
